### ISSUE TITLE
Hubble sometimes panics due to invalid memory address

### ISSUE THREAD

**Issue by dharmab** [cilium/hubble#264]
`kubectl describe`:

```
  Warning  BackOff    2m48s (x705 over 2d1h)  kubelet, cpt-worker-ethos11stageut1-134.ethos.adobe.net  Back-off restarting failed container
```

`kubectl logs`:

```
time="2020-05-26T16:53:24Z" level=info msg="Configured metrics plugin" name=dns status=
time="2020-05-26T16:53:24Z" level=info msg="Configured metrics plugin" name=drop status=
time="2020-05-26T16:53:24Z" level=info msg="Configured metrics plugin" name=tcp status=
time="2020-05-26T16:53:24Z" level=info msg="Configured metrics plugin" name=flow status=
time="2020-05-26T16:53:24Z" level=info msg="Configured metrics plugin" name=port-distribution status=
time="2020-05-26T16:53:24Z" level=info msg="Configured metrics plugin" name=icmp status=
time="2020-05-26T16:53:24Z" level=info msg="Configured metrics plugin" name=http status=
time="2020-05-26T16:53:24Z" level=info msg="Configuring Hubble server" eventQueueSize=1024 maxFlows=131071
time="2020-05-26T16:53:24Z" level=info msg="Starting gRPC server on client-listener" client-listener="0.0.0.0:50051"
time="2020-05-26T16:53:24Z" level=info msg="Starting gRPC server on client-listener" client-listener="unix:///var/run/hubble.sock"
time="2020-05-26T16:53:24Z" level=warning msg="Failed to obtain IPCache from Cilium. If you are using Cilium 1.6 or older, this is expected. Pod names of endpoints running on remote nodes will not be resolved."
time="2020-05-26T16:54:12Z" level=error msg="Updated or created endpoint not found!" error="[GET /endpoint/{id}][404] getEndpointIdNotFound " id=1665
time="2020-05-26T16:56:18Z" level=error msg="Updated or created endpoint not found!" error="[GET /endpoint/{id}][404] getEndpointIdNotFound " id=3494
time="2020-05-26T16:56:19Z" level=error msg="Updated or created endpoint not found!" error="[GET /endpoint/{id}][404] getEndpointIdNotFound " id=3494
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xc7443f]

goroutine 37 [running]:
github.com/cilium/hubble/pkg/api/v1.(*Endpoint).EqualsByID(...)
	/go/src/github.com/cilium/hubble/pkg/api/v1/endpoint.go:35
github.com/cilium/hubble/pkg/api/v1.(*Endpoints).SyncEndpoints(0xc000199ec0, 0xc00a27eb00, 0x13, 0x20)
	/go/src/github.com/cilium/hubble/pkg/api/v1/endpoint.go:80 +0xdf
github.com/cilium/hubble/pkg/cilium.(*State).syncEndpoints(0xc0004047e0)
	/go/src/github.com/cilium/hubble/pkg/cilium/endpoint.go:67 +0x786
created by github.com/cilium/hubble/pkg/cilium.(*State).Start
	/go/src/github.com/cilium/hubble/pkg/cilium/state.go:79 +0x43
```

**Comment by Rolinh** [comment]
Which version are you running? This has been fixed in v0.5.1.

**Comment by dharmab** [comment]
0.5.0. I did check the releases tab but the 0.5.1 notes didn't mention this: https://github.com/cilium/hubble/releases/tag/v0.5.1

I'll update to 0.5.1!

**Comment by Rolinh** [comment]
Yes, we indeed don't have proper release notes. I'm pretty sure your report is a duplicate of #197 which is fixed in v0.5.1
